From cf32fbbe89846b581cfdb80c176aa73e2114714a Mon Sep 17 00:00:00 2001
From: Alexandros Frantzis <alexandros.frantzis@collabora.com>
Date: Fri, 8 Apr 2022 11:03:16 +0300
Subject: [PATCH 114/170] winewayland.drv: Expose
 wayland_shm_buffer_create_from_native.

This will be need for the upcoming cross-process rendering functionality.

Signed-off-by: Alexandros Frantzis <alexandros.frantzis@collabora.com>
---
 dlls/winewayland.drv/wayland_shm.c | 9 +++++++--
 dlls/winewayland.drv/waylanddrv.h  | 2 ++
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_shm.c b/dlls/winewayland.drv/wayland_shm.c
index 254f574f33b..ef2b65abb2d 100644
--- a/dlls/winewayland.drv/wayland_shm.c
+++ b/dlls/winewayland.drv/wayland_shm.c
@@ -35,8 +35,13 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(waylanddrv);
 
-static struct wayland_shm_buffer *wayland_shm_buffer_create_from_native(struct wayland *wayland,
-                                                                        struct wayland_native_buffer *native)
+/**********************************************************************
+ *          wayland_shm_buffer_create_from_native
+ *
+ * Creates a wayland SHM buffer from the specified native buffer.
+ */
+struct wayland_shm_buffer *wayland_shm_buffer_create_from_native(struct wayland *wayland,
+                                                                 struct wayland_native_buffer *native)
 {
     struct wayland_shm_buffer *shm_buffer;
     struct wl_shm_pool *pool;
diff --git a/dlls/winewayland.drv/waylanddrv.h b/dlls/winewayland.drv/waylanddrv.h
index cc47e493ae6..649f5432b57 100644
--- a/dlls/winewayland.drv/waylanddrv.h
+++ b/dlls/winewayland.drv/waylanddrv.h
@@ -449,6 +449,8 @@ void wayland_native_buffer_deinit(struct wayland_native_buffer *native) DECLSPEC
  *          Wayland SHM buffer
  */
 
+struct wayland_shm_buffer *wayland_shm_buffer_create_from_native(struct wayland *wayland,
+                                                                 struct wayland_native_buffer *native) DECLSPEC_HIDDEN;
 struct wayland_shm_buffer *wayland_shm_buffer_create(struct wayland *wayland,
                                                      int width, int height,
                                                      enum wl_shm_format format) DECLSPEC_HIDDEN;
-- 
2.36.1

